bundle/config/mutator/trampoline.go

```diff
@@ -18,22 +18,37 @@ type TaskWithJobKey struct {
 }
 
 type TrampolineFunctions interface {
-	GetTemplateData(task *jobs.Task) (map[string]any, error)
+	GetTemplateData(b *bundle.Bundle, task *jobs.Task) (map[string]any, error)
 	GetTasks(b *bundle.Bundle) []TaskWithJobKey
+	GetTemplate(b *bundle.Bundle, task *jobs.Task) (string, error)
 	CleanUp(task *jobs.Task) error
 }
 type trampoline struct {
 	name      string
 	functions TrampolineFunctions
-	template  string
 }
 
 func NewTrampoline(
 	name string,
 	functions TrampolineFunctions,
-	template string,
 ) *trampoline {
-	return &trampoline{name, functions, template}
+	return &trampoline{name, functions}
+}
+
+func GetTasksWithJobKeyBy(b *bundle.Bundle, filter func(*jobs.Task) bool) []TaskWithJobKey {
```

Comment:
nit: I'd move it move to some other place, making a call `mutator. GetTasksWithJobKeyBy ` is a bit misleading